pyproject.toml

```diff
@@ -1,17 +1,87 @@
+[tool.poetry]
+name = "hera"
+version = "2.6.1"
```

Comment:
I naively implemented `version`. I prefer using [poetry-dynamic-versioning](https://github.com/mtkennerly/poetry-dynamic-versioning) to add version by `git tag`.

@flaviuvadan Do you have any ideas on this?

Reply:
I quite like the verbosity of `VERSION` thus far but I am definitely open to other approaches! Does the version need to be specified here? I wonder if we can leave the `setup.py` as it is?

Reply:
> I quite like the verbosity of VERSION

Yeah, I understand this 👍. This is very simple and easy to understand what the contributors should do. 
Currently we can't use VERSION file on pyproject.toml (pyproject.toml is `toml` so we could not load file dynamically).

> I think it's ok to leave this as is but update the contribution guideline to suggest that users increment this file rather than VERSION.

I updated this [here](https://github.com/argoproj-labs/hera-workflows/pull/94/files#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055R15)